utils/python/CIME/XML/generic_xml.py

```diff
@@ -230,11 +240,9 @@ def get_resolved_value(self, raw_value):
             elif var == "SRCROOT":
                 srcroot = os.path.join(get_cime_root(),"..")
                 item_data = item_data.replace(m.group(), srcroot)
-            elif var in os.environ:
```

Comment:
There may be unintended consequences here, but I can't think of any specific cases right now.